slicebase/bin/rsyncd_config.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python
+"""Generates an rsyncd configuration."""
+
+import optparse
+import sys
+
+ALLOW_HOSTS = ['108.170.192.0/18', '108.177.0.0/20', '142.250.0.0/15',
+               '172.217.0.0/16', '172.253.0.0/16', '173.194.0.0/16',
+               '192.178.0.0/15', '199.87.241.32/28', '207.223.160.0/20',
+               '209.85.128.0/17', '216.239.32.0/19', '216.58.192.0/19',
+               '64.233.160.0/19', '66.102.0.0/20', '66.249.64.0/19',
+               '70.32.128.0/19', '70.90.219.48/29', '70.90.219.72/29',
+               '72.14.192.0/18', '74.125.0.0/16', '23.228.128.64/26']
+
+RSYNC_HEADER = """
+pid file = /var/run/rsyncd.pid
+port = {port}
+hosts allow = {allow_hosts}
+"""
+
+MODULE_TEMPLATE = """
+[{module}]
+    comment = Data from {module}: See http://www.measurementlab.net
+    path = {directory}
+    list = yes
+    read only = yes
+    transfer logging = no
+    ignore errors = no
+    ignore nonreadable = yes
+"""
+
+
+def usage():
+    return """
+DESCRIPTION:
+    rsyncd_config.py generates rsyncd configuration files for M-Lab experiments.
+
+EXAMPLES:
+
+    Standard slice configuration:
+
+        rsyncd_config.py \\
+            --module mlab_slicename \\
+            --directory /var/spool/mlab_slicename
```

Comment:
The directory prefix will always be `/var/spool/<slicename>/<somedir>`.  Instead of an absolute path here, should it just be a name like `--directory SideStream`, which would result in:

`path = /var/spool/<slicename>/SideStream`